--- lib/python2.7/site-packages/tw2/forms/widgets.py	2014-07-27 18:36:55.334163611 +0000
+++ lib/python2.7/site-packages/tw2/forms/widgets.py	2014-07-27 18:39:04.675320805 +0000
@@ -39,19 +39,19 @@
 
     value = twc.Param(attribute=True)
 
-    required = twc.Param('Input field is required',
-        attribute=True, default=None)
+#    required = twc.Param('Input field is required',
+#        attribute=True, default=None)
 
     autofocus = twc.Param('Autofocus form field (HTML5 only)',
         attribute=True, default=None)
 
     template = "tw2.forms.templates.input_field"
 
-    def prepare(self):
-        super(InputField, self).prepare()
-        self.safe_modify('attrs')
-        self.attrs['required'] = 'required' if self.required in [True, 'required'] else None
-        self.required = None  # Needed because self.required would otherwise overwrite self.attrs['required'] again
+#    def prepare(self):
+#        super(InputField, self).prepare()
+#        self.safe_modify('attrs')
+#        self.attrs['required'] = 'required' if self.required in [True, 'required'] else None
+#        self.required = None  # Needed because self.required would otherwise overwrite self.attrs['required'] again
 
 
 class PostlabeledInputField(InputField):
@@ -731,7 +731,7 @@
             return bool(val)
 
     def to_python(self, value, state=None):
-        return [v for v in value if self.any_content(v)]
+        return [v for v in value or [] if self.any_content(v)]
 
 
 class GridLayout(twc.RepeatingWidget):
